pkg/codefresh/project_test.go

```diff
@@ -0,0 +1,25 @@
+package codefresh
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var clientOptions = ClientOptions{
+	Auth: AuthOptions{
+		Token: "",
+	},
+	Debug: true,
+	Host:  "https://g.codefresh.io",
+}
+
+func TestProject(t *testing.T) {
+
+	cf := New(&clientOptions)
```

Comment:
maybe you can use the codefresh/mock/codefresh struct to make this test better?
you will also need to regenerate the mock itself to include your `Projects()` func. You will need mockery 1.1.1 installed, and run `go generate ./...` for it.
let me know if you need any help with it.

Reply:
@noam-codefresh is there an example that shows the coddfresh mock structs use.

For an API client that should have been code generated, this abstraction is proving to be rather tedious, and hard to follow, it most certainly confuses the compiler